### PR TITLE
Add generalized Conv+Activation MKLDNN fuse pass creation Part2

### PR DESCRIPTION
**Pull request by grygielski**
Second part of https://github.com/PaddlePaddle/Paddle/pull/18942.
Changes since Part1:
- deleted old code in graph_pattern_detector
- deleted old pass files
- improvements suggested by Sand3r-
- code introduced in https://github.com/PaddlePaddle/Paddle/pull/18988 has been adjusted to current mkldnn conv+activation pass code

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc [hunk 1]
```diff
@@ -62,14 +62,10 @@ void ConvActivationFusePass::ApplyImpl(ir::Graph* graph) const {
                     boost::get<float>(activation->Op()->GetAttr("threshold")));
     } else {
       desc->SetAttr("fuse_alpha",
-                    activation->Op()->HasAttr("alpha")
-                        ? boost::get<float>(activation->Op()->GetAttr("alpha"))
-                        : 0.0f);
+                    activation->Op()->GetAttrIfExists<float>("alpha"));
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc:65]: Here `float` yields undefined behavior.

File: paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc [hunk 2]
```diff
@@ -62,14 +62,10 @@ void ConvActivationFusePass::ApplyImpl(ir::Graph* graph) const {
...
     }
     desc->SetAttr("fuse_beta",
-                  activation->Op()->HasAttr("beta")
-                      ? boost::get<float>(activation->Op()->GetAttr("beta"))
-                      : 0.0f);
+                  activation->Op()->GetAttrIfExists<float>("beta"));
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc:68]: "

File: paddle/fluid/framework/op_desc.h [hunk 3]
```diff
@@ -80,6 +80,15 @@ class OpDesc {
 
   Attribute GetAttr(const std::string &name) const;
 
+  template <typename T>
+  T GetAttrIfExists(const std::string &name) const {
+    T result;
```
Comment (wojtuss) [inline paddle/fluid/framework/op_desc.h]: This causes undefined behavior for built-in data types.

**Review by wojtuss — CHANGES_REQUESTED** [review]

File: paddle/fluid/framework/op_desc.h (hunk 3 above)
Comment (grygielski) [reply paddle/fluid/framework/op_desc.h]: Great observation, thanks!

**Review by wojtuss — COMMENTED** [review]
LGTM

**Review by wojtuss — APPROVED** [review]

**Comment by wojtuss** [comment]
@luotao1
Could you please help review this PR? It is required for the PR_CI_CPU_Py2 (Compile) build to pass.

**Review by luotao1 — APPROVED** [review]

**Pull request merged**